broker.go

```diff
@@ -82,6 +84,38 @@ func (b *Broker) Open(conf *Config) error {
 		b.conn = newBufConn(b.conn)
 
 		b.conf = conf
+
+		if conf.Net.SASL.Enable {
+			//
+			// Begin SASL/PLAIN authentication
+			//
+			authBytes := []byte("\x00" + b.conf.Net.SASL.User + "\x00" + b.conf.Net.SASL.Password)
+			buf := new(bytes.Buffer)
+
+			err = binary.Write(buf, binary.BigEndian, int32(len(authBytes)))
+			if err != nil {
+				Logger.Printf("Failed to encode payload size (SASL credentials): %s", err.Error())
+			}
+
+			err = binary.Write(buf, binary.BigEndian, authBytes)
+			if err != nil {
+				Logger.Printf("Failed to encode payload (SASL credentials): %s", err.Error())
+			}
+
+			b.conn.SetWriteDeadline(time.Now().Add(b.conf.Net.WriteTimeout))
+			b.conn.Write(buf.Bytes())
+
+			header := make([]byte, 4)
+			n, err := io.ReadFull(b.conn, header)
+			if err != nil {
+				Logger.Printf("Failed to read response while authenticating with SASL: %s", err.Error())
+			}
+			Logger.Printf("SASL authentication successful:\n%v\n%v\n%v", n, header, string(header))
```

Comment:
I don't know a whole lot about SASL but don't you need to actually *check* the response header in some way before saying it succeeded?